runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java

```diff
@@ -271,27 +211,40 @@ public MetricResults metrics() {
   @Override
   public <T> Iterable<T> get(PCollection<T> pcollection) {
     @SuppressWarnings("unchecked")
-    RDDHolder<T> rddHolder = (RDDHolder<T>) pcollections.get(pcollection);
-    Iterable<WindowedValue<T>> windowedValues = rddHolder.getValues(pcollection);
+    BoundedDataset<T> boundedDataset = (BoundedDataset<T>) datasets.get(pcollection);
+    Iterable<WindowedValue<T>> windowedValues = boundedDataset.getValues(pcollection);
     return Iterables.transform(windowedValues, WindowingHelpers.<T>unwindowValueFunction());
   }
 
   <T> Iterable<WindowedValue<T>> getWindowedValues(PCollection<T> pcollection) {
     @SuppressWarnings("unchecked")
-    RDDHolder<T> rddHolder = (RDDHolder<T>) pcollections.get(pcollection);
-    return rddHolder.getValues(pcollection);
+    BoundedDataset<T> boundedDataset = (BoundedDataset<T>) datasets.get(pcollection);
+    return boundedDataset.getValues(pcollection);
   }
 
   @Override
   public void close(boolean gracefully) {
-    // graceful stop is used for streaming.
+    if (jssc != null) {
+      // stop streaming context
+      if (timeout > 0) {
+        jssc.awaitTerminationOrTimeout(timeout);
+      } else {
+        jssc.awaitTermination();
+      }
+      // stop streaming context gracefully, so checkpointing (and other computations) get to
+      // finish before shutdown.
+      jssc.stop(false, gracefully);
+    }
+    state = State.DONE;
     SparkContextFactory.stopSparkContext(jsc);
   }
 
-  /** The runner is blocking. */
+  /**
```

Comment:
The comment `the runner is blocking` is not true anymore in this context. Remove it.